app/models/review.rb

```diff
@@ -1,4 +1,8 @@
 class Review < ApplicationRecord
   belongs_to :user
   belongs_to :chalet
+
+  validates :content, :rating, :user, :chalet, presence: true
+  validates :rating, inclusion: { in: (1..5) }
+  validates :chalet, uniqueness: { scope: :user, message: "Only one review per user about this chalet" }
```

Comment:
Love the message bit! Can we make it more user-friendly? e.g. "You can only review this chalet once." Users don't like to be called users 👀